common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManagerSimpleCacheImpl.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * A simple metadata store definition that uses SharedPreferencesFileManager to persist, read,
+ * update, and delete data. Please note that all CRUD actions return success, as the underlying
+ * store's API does not surface a success indicator. If you need stronger guarantees that an
+ * operation was successful, use {@link SharedPreferencesSimpleCacheImpl} which is less performance
+ * oriented.
+ * <p>
+ * Data serializes as JSON.
+ *
+ * @param <T> The type of metadata that will be persisted.
+ * @see SharedPreferencesSimpleCacheImpl
+ */
+public abstract class SharedPreferencesFileManagerSimpleCacheImpl<T> implements ISimpleCache<T>, IListTypeToken {
+
+    private static final String TAG = SharedPreferencesFileManagerSimpleCacheImpl.class.getSimpleName();
+    private static final String EMTPY_ARRAY = "[]";
+    private static final String TIMING_TAG = "execWithTiming";
+
+    private final ISharedPreferencesFileManager mSharedPrefsFileManager;
+    private final String mKeySingleEntry;
+    private final Gson mGson = new Gson();
+
+    /**
+     * Constructs a new SharedPreferencesFileManagerSimpleCacheImpl. Convenience class for persisting
+     * lists of arbitrarily-typed data.
+     *
+     * @param context   The current app's {@link Context}.
+     * @param prefsName The name of the underlying {@link android.content.SharedPreferences} file.
+     * @param singleKey The name of the key under which all entries will be cached.
+     */
+    public SharedPreferencesFileManagerSimpleCacheImpl(@NonNull final Context context,
+                                                       @NonNull final String prefsName,
+                                                       @NonNull final String singleKey) {
+        this(SharedPreferencesFileManager.getSharedPreferences(
+                context,
+                prefsName,
+                Context.MODE_PRIVATE,
+                null // File is not encrypted
+                ),
+                singleKey
+        );
+    }
+
+    /**
+     * Constructs a new SharedPreferencesFileManagerSimpleCacheImpl from the provided
+     * {@link SharedPreferencesFileManager}, using the provided singleKey for the underlying collection.
+     *
+     * @param sharedPreferencesFileManager The underlying {@link SharedPreferencesFileManager} to use.
+     * @param singleKey                    The name of the key under which all entries will be cached.
+     */
+    public SharedPreferencesFileManagerSimpleCacheImpl(@NonNull final SharedPreferencesFileManager sharedPreferencesFileManager,
+                                                       @NonNull final String singleKey) {
+        Logger.verbose(TAG + "::ctor", "Init");
+        mSharedPrefsFileManager = sharedPreferencesFileManager;
+        mKeySingleEntry = singleKey;
+    }
+
+    private interface NamedRunnable<V> extends Callable<V> {
+        String getName();
+    }
+
+    private <V> V execWithTiming(@NonNull final NamedRunnable<V> runnable) {
```

Comment:
Wondering whether this implementation should be in a separate class ... perhaps accessed via a Timing interface?

Reply:
Also wondering if we should be able to enable/disable timing via a flag.